src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonResource.kt

```diff
@@ -72,4 +77,65 @@ class PrisonPersonResource(
   fun getPhysicalAttributes(
     @Schema(description = "Offender number", example = "A1234AA") @PathVariable offenderNo: String,
   ): PrisonerPhysicalAttributesResponse = service.getPhysicalAttributes(offenderNo)
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_PRISON_PERSON')")
+  @PutMapping("/prisoners/{offenderNo}/physical-attributes")
+  @Operation(
+    summary = "Upsert physical attributes for a prisoner",
+    description = "Upserts physical attributes on the latest booking for a prisoner, if it exists. Requires ROLE_NOMIS_PRISON_PERSON",
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = UpsertPhysicalAttributesRequest::class),
+        ),
+      ],
+    ),
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Physical Attributes Updated",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = UpsertPhysicalAttributesResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden to access this endpoint. Requires ROLE_NOMIS_PRISON_PERSON",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Prisoner does not exist or has no bookings",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun upsertPhysicalAttributes(
+    @Schema(description = "Offender number", example = "A1234AA") @PathVariable offenderNo: String,
+    @RequestBody @Valid request: UpsertPhysicalAttributesRequest,
+  ): UpsertPhysicalAttributesResponse = service.upsertPhysicalAttributes(offenderNo, request)
```

Comment:
This is the main change - put an endpoint in front of the service and return a response.